From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct tcp_sock
Patch-mainline: never, kabi
References: kabi

In 4.4.136, commit 70741861fc4389ba1228ae4f24cf816b2a79fde6 (tcp: avoid
integer overflows in tcp_rcv_space_adjust()), upstream commit
607065bad9931e72207b0cac365d7d4abc06bd99 changed int space to u32. It
made the kABI checker to complain.

Given the size of the type did not change and only the interpretation
did, just hide the change from the kABI checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/tcp.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/linux/tcp.h
+++ b/include/linux/tcp.h
@@ -324,7 +324,11 @@ struct tcp_sock {
 
 /* Receiver queue space */
 	struct {
+#ifdef __GENKSYMS__
+		int	space;
+#else
 		u32	space;
+#endif
 		u32	seq;
 		u32	time;
 	} rcvq_space;
